src/reporters.ts

```diff
@@ -104,6 +104,17 @@ export const reportUseSelf = ({ markers, lines, lineIndex, value }: ReporterOpts
   });
 };
 
+export const reportInvalidName = ({ markers, lines, lineIndex, value, clause }: ReporterOpts) => {
```

Comment:
This should probably be more specific. It should either say type XYZ or relation ABC of type XYZ does not match..

Reply:
Done